mne/source_estimate.py

```diff
@@ -2288,15 +2489,34 @@ def morph_data_precomputed(subject_from, subject_to, stc_from, vertices_to,
     if not sum(len(v) for v in vertices_to) == morph_mat.shape[0]:
         raise ValueError('number of vertices in vertices_to must match '
                          'morph_mat.shape[0]')
+
     if not stc_from.data.shape[0] == morph_mat.shape[1]:
         raise ValueError('stc_from.data.shape[0] must be the same as '
                          'morph_mat.shape[0]')
 
     if stc_from.subject is not None and stc_from.subject != subject_from:
         raise ValueError('stc_from.subject and subject_from must match')
-    data = morph_mat * stc_from.data
-    stc_to = SourceEstimate(data, vertices_to, stc_from.tmin, stc_from.tstep,
-                            verbose=stc_from.verbose, subject=subject_to)
+
+    if isinstance(stc_from, VectorSourceEstimate):
+        # Morph the locations of the dipoles, but not their orientation
+        morph_xyz = lil_matrix((morph_mat.shape[0] * 3,
+                                morph_mat.shape[1] * 3))
+        morph_xyz[0::3, 0::3] = morph_mat
+        morph_xyz[1::3, 1::3] = morph_mat
+        morph_xyz[2::3, 2::3] = morph_mat
+        morph_xyz = morph_xyz.tocsr()
```

Comment:
I think it would be faster and more memory efficient to do:
```
data = np.empty(...)
data[::3] = morph * stc_from.data[:, 0]
data[1::3] = morph ...
```